### PR TITLE
Fix win build

### PR DESCRIPTION
**Pull request by slobodin**
std::max and win32 max macro

### CODE REVIEW

File: cocos2dx/cocoa/CCGeometry.cpp [hunk 1]
```diff
@@ -303,10 +303,12 @@ Rect Rect::unionWithRect(const Rect & rect) const
...
-    float combinedLeftX = std::min(thisLeftX, otherLeftX);
-    float combinedRightX = std::max(thisRightX, otherRightX);
-    float combinedTopY = std::max(thisTopY, otherTopY);
-    float combinedBottomY = std::min(thisBottomY, otherBottomY);
+    using namespace std;
+    
+    float combinedLeftX = min(thisLeftX, otherLeftX);
+    float combinedRightX = max(thisRightX, otherRightX);
+    float combinedTopY = max(thisTopY, otherTopY);
+    float combinedBottomY = min(thisBottomY, otherBottomY);
```
Comment (dumganhar) [inline cocos2dx/cocoa/CCGeometry.cpp:311]: Can't `std::min`, `std::max`  work on windows?

**Comment by slobodin** [comment]
They can't because they hiding by WinAPI macro definition `max` and `min`. Until we specify `NOMINMAX` define before including `windows.h`

**Comment by winipcfg** [comment]
FYI:
http://stackoverflow.com/questions/5004858/stdmin-gives-error

However, adding statement "using namespace std" is not a good practice in general
http://stackoverflow.com/questions/1452721/why-is-using-namespace-std-considered-bad-practice

**Comment by slobodin** [comment]
> adding statement "using namespace std" is not a good practice in general

In this case this statement takes effect only in scope of `Rect::unionWithRect` function.

So, in windows it uses WinAPI macro defs, in gcc it uses `std::min` or `std::max`.

**Comment by dumganhar** [comment]
Fixed at https://github.com/cocos2d/cocos2d-x/pull/3090

**Comment by dumganhar** [comment]
Thanks.

**Comment by dumganhar** [comment]
Now we're supporting 'std::max' and 'MAX' at the same time.

**Pull request closed without merging**